### ISSUE TITLE
Incorrect readlink usage on OS X

### ISSUE THREAD

**Issue by hameerabbasi** [slacka/WinUSB#63]
Hello. No matter what I do, I get exactly the following output on OS X (the program compiled and installed fine):
```
readlink: illegal option -- f
usage: readlink [-n] [file ...]
usage: basename string [suffix]
       basename [-a] [-s suffix] string [...]
```

`readlink` seems to work fine otherwise:
```
$ readlink HexChat 
/usr/local/bin/hexchat
```

**Comment by Holzhaus** [comment]
Looks like that option is only present in the GNU coreutils version of readlink:

http://stackoverflow.com/questions/1055671/how-can-i-get-the-behavior-of-gnus-readlink-f-on-a-mac

**Comment by Lin-Buo-Ren** [comment]
I wonder if we do need the `-f` switch...

**Comment by hameerabbasi** [comment]
So, as I see it, there are two options.

1. If we don't need `-f`, just remove it.
2. If we do need `-f`, include a version/implementation of `readlink`.

**Comment by Lin-Buo-Ren** [comment]
I personally prefer 1., another (much complex) way to solve this problem is to port the `winusb` bash script to Python/Ruby/Insert your favorite programming language.

**Comment by slacka** [comment]
Does the [python port of WinUSB](https://github.com/Mati365/pyWinUSB) work?

**Comment by hameerabbasi** [comment]
I had to go through a lot just to get that installed. Python3 (native on OS X) didn't have `gi`, had to go through `brew` with [this](http://stackoverflow.com/a/35685336). Even after that, I got a bunch of errors, most prominently this:
```
grep: /sys/block/sd*/removable: No such file or directory
```
(designed for the Linux directory structure?) and the fact that it wouldn't even list _files_ in the ISO picker. Only directories. And yes, there was an ISO in the directory.

**Comment by Lin-Buo-Ren** [comment]
@hameerabbasi WinUSB does rely on several Linux sysfs and devfs paths to find available devices and their parameters, so unless Mac OS support such mechanic I'm afraid WinUSB won't run on it(at least not until we replace the calls with something that is cross-platform.

**Comment by hameerabbasi** [comment]
I see. At least configure it so it generates a warning on OS X instead of compiling pretty much fine?

**Comment by slacka** [comment]
We do not have OS X. Patches are welcome.

**Comment by Lin-Buo-Ren** [comment]
> I see. At least configure it so it generates a warning on OS X instead of compiling pretty much fine?

I believe it should be done in the currently incomplete documentation/README to indicate the dependence on Linux sysfs etc., users should read them prior than trying to build on a not-supported platform.

**Comment by slacka** [comment]
commit 8d4bf75 closes #63. Readme now makes it clear that this is for Linux only. Of course, we'll accept any patches to improve OS X support.
